common-npm-packages/azure-arm-rest-v2/azure-arm-storage.ts

```diff
@@ -217,6 +220,41 @@ export class StorageAccounts {
         return "";
     }
 
+    private async getStorageAccountsByUri(uri: string, filterName: string, options?: any): Promise<Model.StorageAccount[]> {
+        const request = new webClient.WebRequest();
+        request.method = 'GET';
+        request.headers = this.client.setCustomHeaders(options);
+        request.uri = uri;
+
+        const storageAccounts: Model.StorageAccount[] = [];
+
+        const response = await this.client.beginRequest(request);
+        if (response.statusCode !== 200) {
+            throw azureServiceClientBase.ToError(response);
+        }
+
+        storageAccounts.push(...response.body.value);
+
+        if (filterName) {
+            const targetSA = storageAccounts.find(sa => sa.name = filterName);
+
+            if (targetSA) {
+                return [targetSA];
+            }
+        }
+
+        if (response.body.nextLink) {
+            const nextResult = await this.client.accumulateResultFromPagedResult(response.body.nextLink);
+            if (nextResult.error) {
+                throw nextResult.error;
+            }
+
+            storageAccounts.push(...nextResult.result);
+        }
+
+        return storageAccounts;
```

Comment:
Same filterName logic should be applied. 

if(filterName)
{
       const targetSA = storageAccounts.find(sa => sa.name === filterName);
         if (targetSA) {
                return [targetSA];
            }
} else {
return storageAccounts;
}